### ISSUE TITLE
Use-before-def is reported incorrectly for switch statement expressions

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#2280]
```TypeScript
var x = 10;

switch (x) {
    case 10:
        let x = 20;
}
```

It is worth ensuring the spec behavior, but I was under the impression that the `x` in `switch (x)` would refer to the outer `x`. Instead, we get

> Block-scoped variable 'x' used before its declaration.
